### PR TITLE
ci: Fix changeset workflow

### PR DESCRIPTION
**Pull request by tylerbutler**
The changeset workflow was not functioning because the changeset CLI didn't work for our branches. This PR updates the workflow to use the `flub check changeset` command which is designed for this purpose.

It's divided into two workflows. The first checks PRs for the presence of a new changeset file. When a PR is opened, it will be checked for a changeset. The results of the check are uploaded as a pipeline artifact. After this workflow is completed it triggers the changeset-reporter workflow, which adds a comment to the PR depending on the results of the check.

The workflows are separated for security reasons. One requires write access to the repo and one does not.

### CODE REVIEW

File: .github/workflows/pr-labeler.yml [hunk 1]
```diff
@@ -24,28 +24,29 @@ jobs:
   changeset_label:
     name: Changeset check
     runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'changeset-required')
     steps:
```
Comment (tylerbutler) [inline .github/workflows/pr-labeler.yml:28]:
```suggestion
    runs-on: ubuntu-latest
    # We want to run this workflow even for PRs that don't require changesets.
    # if: contains(github.event.pull_request.labels.*.name, 'changeset-required')
    steps:
```

File: .github/workflows/pr-labeler.yml [hunk 2]
```diff
@@ -24,28 +24,29 @@ jobs:
...
       - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # ratchet:actions/setup-node@v3
         with:
-          node-version: "16"
+          node-version-file: .nvmrc
+          cache: "pnpm"
+          cache-dependency-path: pnpm-lock.yaml
       - name: Install tools
         run: |
-          pnpm add -g @changesets/cli@2.26.0 changesets-format-with-issue-links@0.3.0
+          pnpm install -w --frozen-lockfile
```
Comment (tylerbutler) [inline .github/workflows/pr-labeler.yml:42]:
```suggestion
          # We only need to install the root dependencies
          pnpm install -w --frozen-lockfile
```

File: .github/workflows/changeset-reporter.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    workflows: [ "pr-check-changeset" ]
+    types:
+      - completed
+
+jobs:
+  load_report:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download results
+        uses: dawidd6/action-download-artifact@bd10f381a96414ce2b13a11bfa89902ba7cea07f # ratchet:dawidd6/action-download-artifact@v2.24.3
```
Comment (alexvy86) [inline .github/workflows/changeset-reporter.yml:19]: There's no official action to download an artifact like there is for publishing them?

**Review by alexvy86 — APPROVED** [review]

File: .github/workflows/changeset-reporter.yml (hunk 3 above)
Comment (tylerbutler) [reply .github/workflows/changeset-reporter.yml:19]: I know, it's a little unexpected. You can only upload/download artifacts in the same workflow as far as I can tell.

"Let's suppose you have a workflow with a job in it that at the end uploads an artifact using actions/upload-artifact action and you want to download this artifact in another workflow that is run after the first one. Official actions/download-artifact does not allow this."

File: .github/workflows/pr-check-changeset.yml [hunk 4]
```diff
@@ -1,40 +1,79 @@
...
+
+      # Add the remote in forked repos so we can more easily test in forks
+      - run: git remote add upstream https://github.com/microsoft/FluidFramework
+        if: github.repository_owner != 'microsoft'
+      - run: git fetch upstream
+        if: github.repository_owner != 'microsoft'
+      - run: git log -1
+        if: github.repository_owner != 'microsoft'
+
+      # Check whether a changeset was added. This step will have the outcome 'failure' if there is no changeset.
```
Comment (tylerbutler) [inline .github/workflows/pr-check-changeset.yml:47]:
```suggestion
      # Check whether a changeset was added. This step will have the outcome '1' if there is no changeset.
```

**Pull request merged**